rbac/management/principal/it_service.py

```diff
@@ -303,11 +303,34 @@ def get_service_accounts(self, user: User, options: dict[str, Any] = {}) -> Tupl
         offset = options.get("offset")
         limit = options.get("limit")
         limit_offset_validation(offset, limit)
+        # Service account filtering query parameters
+        name = options.get("name")
+        owner = options.get("owner")
+        description = options.get("description")
+        filtered_service_accounts = []
+        sa_query_passed = name or owner or description
 
         count = len(service_accounts)
         # flake8 ignore E203 = Whitespace before ':' -> false positive https://github.com/PyCQA/pycodestyle/issues/373
         service_accounts = service_accounts[offset : offset + limit]  # type: ignore # noqa: E203
-
+        # If any one service account filter parameter is provided extract & return the specific service accounts
+        if sa_query_passed:
```

Comment:
I did few tests and it looks like the pagination is not working correctly, my local tenant contains 4 service accounts, 3 of them  belongs to ownerA and the last one to ownerB, I tried to filter by `?owner=ownerA` and I received 3 SA in the response as expected, then I set `limit=2` to check pagination and my response contains only 1 SA (2 expected) and the count under the meta shows 1 (3 is expected), I can provide more details if needed, I believe it is because the limit and offset are set before the filtering is done, I am happy to provide more details if needed